parlai/agents/seq2seq/seq2seq.py

```diff
@@ -357,7 +357,7 @@ def predict(self, xs, ys=None, cands=None, valid_cands=None, lm=False):
             loss.backward()
             self.update_params()
             losskey = 'loss' if not lm else 'lmloss'
-            loss_dict = {losskey: loss.mul_(len(xs)).data}
+            loss_dict = {losskey: loss.mul_(len(xs)).data[0]}
```

Comment:
actually this was intentionally returning the cuda tensor so that we didn't have to pull the tensor back from the GPU, it's possible it's not affecting the performance much though

Reply:
That makes sense, however I think there is an issue with some python built-ins when using cuda tensors; here's the error I get if I revert this change:

Traceback (most recent call last):
File "examples/train_model.py", line 220, in <module>
main(setup_args())
File "examples/train_model.py", line 157, in main
train_report = world.report()
File "/data/users/kshuster/ParlAI/parlai/core/worlds.py", line 645, in report
return self.world.report()
File "/data/users/kshuster/ParlAI/parlai/core/worlds.py", line 263, in report
return self.agents[0].report()
File "/data/users/kshuster/ParlAI/parlai/core/agents.py", line 140, in report                                                                            
return self.metrics.report()
File "/data/users/kshuster/ParlAI/parlai/core/metrics.py", line 181, in report                                                                              
m[k] = round_sigfigs(self.metrics[k] / total, 3)
File "/data/users/kshuster/ParlAI/parlai/core/utils.py", line 91, in round_sigfigs
return round(x, -math.floor(math.log10(abs(x)) - sigfigs + 1))

TypeError: bad operand type for abs(): 'torch.cuda.FloatTensor'

Reply:
yeah actually it looks like it has a pretty minor effect so let's land it